db/migrate/20231218170631_add_rate_to_percentage_amount_details.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class AddRateToPercentageAmountDetails < ActiveRecord::Migration[7.0]
+  class Fee < ApplicationRecord
+    belongs_to :charge, -> { with_discarded }, optional: true
+  end
+
+  def up
+    percentage_fees = Fee.joins(:charge).merge(Charge.percentage).where.not(amount_details: nil)
+
+    percentage_fees.find_each do |fee|
+      fee.update!(
+        amount_details: fee.amount_details.except!('per_unit_amount').merge(
+          rate: BigDecimal(fee.charge.properties['rate'].to_s),
+        ),
+      )
+    end
+
+    graduated_percentage_fees = Fee.joins(:charge).merge(Charge.graduated_percentage).where.not(amount_details: nil)
+
+    graduated_percentage_fees.find_each do |fee|
+      fee.amount_details['graduated_percentage_ranges'] = fee.amount_details['graduated_percentage_ranges'].tap do |ranges|
```

Comment:
Layout/LineLength: Line is too long. [123/120] (https://rubystyle.guide#max-line-length)